pycbc/events/coinc.py

```diff
@@ -369,8 +369,8 @@ def cluster_coincs(stat, time1, time2, timeslide_id, slide, window, **kwargs):
     else:
         time = 0.5 * (time2 + time1)
 
-    tslide = timeslide_id.astype(numpy.float128)
```

Comment:
I do have a concern here, because I'm not convinced that this code would work correctly if it uses 64-bit floats (float128 was chosen here for a reason). Could we instead have some check here (and in the code below), where we first check for float128, and if this doesn't exist we fall back on longdouble with a warning that this code may not work in this instance.

Reply:
Ian managed to put in words what I was also worried about.

Reply:
According to Xan's link above (about extended precision), the `float128` type *never* provides more precision than `longdouble`. It just pads with zeros to align on a 128 bit boundary. And on some systems `longdouble` has the same precision as simply `double`; C leaves that up to the compiler and hardware.

Reply:
Thanks for clarifying @josh-willis ... I had scanned over the page, but I hadn't properly understood that a type called `float128` wasn't actually 128-bit precision!!

I don't think this entirely removes my concern though. If I've understood correctly, on x86-linux this `tslide` object would be stored with 80-bit precision (currently) and will fail on mac. With the change it would still be 80-bit precision on CIT, but will now work on mac but be stored with 64-bit precision. I guess the question is then: Are there any cases where the 80-bit precision is actually needed (and if not, why did this use `float128` in the first place)? @ahnitz Any insights on this?

Reply:
A small test seems to suggest that `longdouble` seems to have *more* precision on a mac than on CIT???

```
ON MAC
>>> numpy.longdouble(1200300400.100200100)
1200300400.1002002
>>> numpy.longdouble(1200300400.100200300)
1200300400.1002004
>>> numpy.longdouble(1200300400.100200200)
1200300400.1002002
```

```
ON CIT
>>> numpy.longdouble(1200300400.100200100)
1200300400.1002001762
>>> numpy.longdouble(1200300400.100200200)
1200300400.1002001762
>>> numpy.longdouble(1200300400.100200300)
1200300400.1002004147
```

Reply:
@spxiwh I think it is conservative most most purposes we'd use it in, but at the time, my thinking was this. 

Maximum analysis time - 1 yr, time slide number - 1000000, time precision 1e-5s. We lie the timeslides end to end to make the calculation easier. 

This requires about 16-17 digits of precision, which is just beyond what can be done with doubles. However, if you relax the numbers above double should work fine. In particular, this is just for the clustering, so if you don't care about some "fuziness" about the cluster window size then you could probably get away with double in most cases, I think.